packages/samples/headless-react/src/components/result-list/result-list.fn.tsx

```diff
@@ -0,0 +1,64 @@
+import {useEffect, useState, FunctionComponent} from 'react';
+import {
+  buildResultList,
+  ResultList as HeadlessResultList,
+} from '@coveo/headless';
+import {engine} from '../../engine';
+
+interface ResultListProps<T extends {[field: string]: string}> {
+  controller: HeadlessResultList;
+  fieldsToInclude: T;
+}
+
+export const ResultList: FunctionComponent<ResultListProps<{
+  [field: string]: string;
+}>> = (props) => {
+  const {controller} = props;
+  const [state, setState] = useState(controller.state);
+
+  useEffect(() => controller.subscribe(() => setState(controller.state)), []);
+
+  if (!state.results.length) {
+    return <div>No results</div>;
+  }
+
+  return (
+    <div>
+      <ul style={{textAlign: 'left'}}>
+        {state.results.map((result) => (
+          <li key={result.uniqueId}>
+            <article>
+              <h2>
+                <a href={result.clickUri}>{result.title}</a>
+              </h2>
+              <table>
+                <tbody>
+                  {Object.keys(props.fieldsToInclude).map(
```

Comment:
Let's remove the `fieldsToInclude` prop and not render the fields. It makes the sample too complex. Just showing a clickable title (line 32) and the excerpt is more than enough.

Reply:
For only the functional component or both?

Reply:
I would do it for both to keep the samples consistent. If someone toggles between the two, I think they should get the same rendered output